File: IsSubsequence.java
public class IsSubsequence {
  class Solution {
    public boolean isSubsequence(String s, String t) {  
        StringBuilder shorterStringBuilder = new StringBuilder(s);
        for (int i = 0; i < t.length(); i ++) {
            if (shorterStringBuilder.toString() == "") {
                return true;
            }
            if (shorterStringBuilder.charAt(0) == t.charAt(i)) {
               shorterStringBuilder.deleteCharAt(0);
            }
        }
        if (shorterStringBuilder.toString() == "") {
            return true;
        }
        return false;
    }
  }
}

File: issubsequence.cpp
class Solution {
public:
    bool isSubsequence(string s, string t) {
        if (s.size() == 0) {
            return true;
        }

        int candidateIdx = 0;
        for (auto c : t) {
            if (candidateIdx < s.size() && s[candidateIdx] == c) {
                candidateIdx++;
            }

            if (candidateIdx == s.size()) {
                return true;
            } 
        }

        return false;
    }
};
